### ISSUE TITLE
App restarts automatically and appears in notification bar

### ISSUE THREAD

**Issue by Improver2000** [y20k/transistor#274]
Although the app was stopped it restarts automatically and appears in notification bar. Only a total kill can prevent this.

Samsung - Android 9

**Comment by y20k** [comment]
Thanks for the report. Could you please add a short description / steps to reproduce the problem.

**Comment by Improver2000** [comment]
Start app and hear radio. Then stop radio. Then when swiping the app away in the notification bar, it appears again after some time.

**Comment by y20k** [comment]
Okay, Thanks. I will try to reproduce this bug.

Two questions:
1. How much time does it take on average until the notification reappears?
2. Which version of Android are you using?

**Comment by MaMiMa81** [comment]
Same happens with me:

On Lineage OS 14.1 based ROM

Reappears after about 2 hours on average from what I've noticed after app is swiped away from notification area - & that is with the app closed from the overview area.
Have just cleared cache & will report back.

By the way see OP:

> Samsung - Android 9

;-)

**Comment by y20k** [comment]
Thanks. I have a Lineage OS 14.1 Nexus 5 lying around. I will run a test, too. See what the logs have to offer here.

**Comment by y20k** [comment]
@MaMiMa81 
I ran a couple of tests on my Nexus 5 LOS 14.1. No reappearing playback notification here. Not sure what I did different that you. Do you have headphone attached when that happens? Bluetooth or wired?

**Comment by MaMiMa81** [comment]
Evening
No headphones attached - just to be clear it was regularly reappearing back in the notification bar/area but not playing & that was pretty much bang on every 40 mins - I have now reverted back to v.3.2.4 because the constant reappearance was quite frankly getting on my t#ts - please don't get me wrong I am appreciative of your excellent app   & if you wish me to test further I will - I am on a rather ancient Moto G (Mk1) of that's of any help.

**Comment by MaMiMa81** [comment]
Good Afternoon
Just updated to v.4.0.4 as was flagged up as the latest update on F-Droid. Did have a cursory look as the latest code & looked like you had updated the libraries - however when installed this same issue was in evidence unfortunately - reappearing in the notification bar every 40 mins approx. - I've reverted back to v.3.2.4 for the time being.
Regards

**Comment by y20k** [comment]
@MaMiMa81 

I was wondering if you are using the Homescreen Shortcut feature, or the Play Last Station feature or any other face way to start playback?

I have no idea what might be the cause of the behavior, since it never happened on my device. But personally do not use the above mentioned features - and I thought that that might be a possible cause of the problem.

**Comment by Improver2000** [comment]
> @MaMiMa81
> 
> I was wondering if you are using the Homescreen Shortcut feature, or the Play Last Station feature or any other face way to start playback?
> 
> I have no idea what might be the cause of the behavior, since it never happened on my device. But personally do not use the above mentioned features - and I thought that that might be a possible cause of the problem.

Yes!! I use the Play Last Station feature. Might that be the reason? If the notification appears again,  I cannot swipe it away but have to start Transistor again and then it works.

**Comment by MaMiMa81** [comment]
> I was wondering if you are using the Homescreen Shortcut feature, or the Play Last Station feature or any other face way to start playback?
> 
> I have no idea what might be the cause of the behaviour, since it never happened on my device. But personally do not use the above mentioned features - and I thought that that might be a possible cause of the problem.

No I don't use any of those features.

Just uninstalled v.3.2.4, rebooted the phone, installed v.4.0.4 & then added stations from the new in app method.
I played one of the newly added stations briefly then closed the app & removed from it the overview section. When I did this though it reappeared in the notification area & I then cleared it from there (this doesn't happen in v.3.2.4 - it does however briefly flashes on before disappearing though).
Its been over an hour & half now since I closed it hasn't reappeared so it seems the issue may be resolved.
You may wish to try this way of updating if you're having issues - I'll post back later with any updates.

**Comment by MaMiMa81** [comment]
Bad news I'm afraid it was back in the notification area after 4 hours

**Comment by y20k** [comment]
@MaMiMa81 

Are these the steps, you took before the Notification reappeared? I need to know the **exact** steps, to be able to recreate the problem. There is no chance of fixing it, if the problem is not reproduce-able for me.

1. long press on the app icon
2. select "Play last station"
3. Transistor comes to foreground and playback starts
4. tap home button / home gesture
5. Transistor is in background
6. stop playback via Notification
7. swipe away notification
8. swipe away the app using the task switcher
9. tap home button / home gesture

... and after 40 minutes average the Notification reappears.

**Comment by MaMiMa81** [comment]
This is how it went:
After installing  v.4.0.4 I opened the app via the F-Droid installation 'page' & added stations as per my post 2 up.
I started play on a station by pressing the play arrow for that particular station in the app.
I then pressed the main stop button in the app.
I then pressed the overview button in the navigation bar bringing up all open apps & swiped away Transistor.
The radio icon then appears in the notification area & I swipe it away too.
Since this last installation instance the app then reappears on the notification area after 4 hours - indeed it has now just appeared again - which is 4 hours since I swiped it away last time.
HTH

**Comment by MaMiMa81** [comment]
Well its over 9 hours now since I last swiped the app away from the notification area & woke up this morning & its not there.
2 things - I don't know if it did appear & went away again of its own velicion? (I very much doubt it though from past experience) & I the phone was on charge for a good proportion of that time.

**Comment by y20k** [comment]
I think I found the bug. There is one thing I oversaw in your descriptions.

Steps to reproduce were basically ...

* tap Stop button
* swipe away the app in task switcher
* -> app is killed
* -> **notification stays** (or better reappears instantly)
* swipe away notification
* ...

I oversaw, that the notification is not really dismissed, when that app is killed. This is the root problem.

With that in mind I found the problem. Yay. Thanks for staying with me on this bug.
